src/checkpoint/checkpoint_txn.c

```diff
@@ -2576,7 +2611,7 @@ __checkpoint_tree_helper(WT_SESSION_IMPL *session, const char *cfg[])
     if (F_ISSET(btree, WT_BTREE_LOGGED))
         F_CLR(txn, WT_TXN_SHARED_TS_READ);
 
-    ret = __checkpoint_tree(session, true, cfg);
+    ret = __checkpoint_tree(session, true, cfg, /*is_mt = */ true);
```

Comment:
How about saving a flag in the worker session to indicate that it is a checkpoint thread, so that you don't need to pass the flag?